pkg/controllers/jobset_controller.go

```diff
@@ -336,21 +341,52 @@ func (r *JobSetReconciler) deleteJobs(ctx context.Context, js *jobset.JobSet, jo
 				finalErr = err
 				return
 			}
-			log.V(2).Info("successfully deleted job", "job", klog.KObj(job), "restart attempt", job.Labels[job.Labels[jobset.RestartsLabel]])
+			log.V(2).Info("successfully deleted job", "job", klog.KObj(job), "restart attempt", job.Labels[job.Labels[jobset.RestartsKey]])
 		}()
 	}
 	wg.Wait()
 	return finalErr
 }
 
-func (r *JobSetReconciler) constructJobsFromTemplate(js *jobset.JobSet, rjob *jobset.ReplicatedJob, ownedJobs *childJobs) ([]*batchv1.Job, error) {
+// updateStatus updates the status of a JobSet.
+func (r *JobSetReconciler) updateStatus(ctx context.Context, js *jobset.JobSet, eventType, eventReason, eventMsg string) error {
+	if err := r.Status().Update(ctx, js); err != nil {
+		return err
+	}
+	r.Record.Eventf(js, eventType, eventReason, eventMsg)
+	return nil
+}
+
+// TODO: update condition in place if it exists.
```

Comment:
we should do this in a follow up PR, this is important

Reply:
Created #39 and referenced it in this TODO.